src/app/views/register/register.controller.spec.js

```diff
@@ -0,0 +1,97 @@
+import registerControllers from './register.controllers';
+import userRepositoryService from './../../services/userRepositoryService/userRepository.services';
+
+describe('Register controller', () => {
+  let $scope, ionicSideMenuDelegateStub, $state, $ionicPopup, $translate, timeout, userRepository, controller;
+
+  beforeEach(angular.mock.module(registerControllers.name));
+
+  beforeEach(angular.mock.module(userRepositoryService.name));
+
+  beforeEach(inject(($rootScope, $controller, $timeout) => {
+    $scope = $rootScope.$new();
+    ionicSideMenuDelegateStub = {canDragContent: sinon.spy(), isOpenLeft: sinon.spy(), toggleLeft: sinon.spy};
+    $state = {go: sinon.spy()};
+    userRepository = {register: sinon.stub()};
+    timeout = $timeout;
+    $ionicPopup = {alert: sinon.stub()};
+    $translate = {instant: sinon.spy()};
+
+    controller = $controller('RegisterController', {
+      $scope: $scope,
+      $ionicSideMenuDelegate: ionicSideMenuDelegateStub,
+      $timeout: timeout,
+      $ionicPopup: $ionicPopup,
+      $translate: $translate,
+      $state: $state,
+      userRepository: userRepository
+    });
+
+    controller.form = {$valid: true};
+  }));
+
+  it('should set up side menu events', () => {
+    $scope.$emit('$ionicView.afterEnter');
+    expect(ionicSideMenuDelegateStub.canDragContent.withArgs(false)).to.be.calledOnce;
+    $scope.$emit('$ionicView.beforeLeave');
+    expect(ionicSideMenuDelegateStub.canDragContent.withArgs(true)).to.be.calledOnce;
+  });
+
+  it('on submit should not try to register when form invalid', () => {
+    controller.form.$valid = false;
+
+    controller.onSubmit();
+
+    expect(userRepository.register).to.not.be.called;
+  });
+
+  it('on submit should show alert when no internet', done => {
+    userRepository.register.returns(Promise.reject(new Error('internet')));
+
+    const onSubmitPromise = controller.onSubmit();
+
+    onSubmitPromise.then(() => {
+      expect($ionicPopup.alert).to.be.calledOnce;
+      done();
+    });
+  });
+
+  it('on submit should show error and empty password when register failed', done => {
```

Comment:
No encaja el it con el nombre del test, lo comentamos

Reply:
Cambiado